doc/api/next_api_changes/development/30767-MS.rst

```diff
@@ -0,0 +1,18 @@
+``legend.linewidth`` rcParam and parameter
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+A new rcParam ``legend.linewidth`` has been added to control the line width of
+the legend's box edges. When set to ``None`` (the default), it inherits the
+value from ``axes.linewidth``. This allows for independent control of the
+legend frame line width without affecting other elements.
+
+.. versionadded:: 3.10
```

Comment:
The versionadded directive is not needed in the release notes.  All these files will later be combined to make a single page for the release.  [Like this](https://matplotlib.org/devdocs/release/prev_whats_new/whats_new_3.10.0.html)